feature/bgp/gracefulrestart/ate_tests/err/README.md

```diff
@@ -0,0 +1,289 @@
+  * Start traffic. Send `gNOI.killProcessRequest_Signal_KILL` as per `gNOI proto` to ATE:Port1 to stop its BGP process abruptly. Configure ATE:Port1 to delay the BGP reestablishment for 330 secs over the Hold-time. Expected behavior in this case is the same as RT-1.35.2.a
+  * Post 330Secs over Hold-time expiry, BGP on ATE:Port1 is expected to be up and all traffic is expected to be successful.<br>
+  * Repeat the same test on ATE:Port2
+  
+  ***`b. DUT Helper for a restarting EBGP speaker whose BGP process was killed abruptly after ERR policy was removed.`***
+  * Start traffic. Send `gNOI.killProcessRequest_Signal_KILL` as per `gNOI proto` to ATE:Port1 to stop its BGP process abruptly. Configure ATE:Port1 to delay the BGP reestablishment for 330 secs over the Hold-time. Expected behavior in this case is the same as RT-1.35.2.b
+  * Post 330Secs over Hold-time expiry, BGP on ATE:Port1 is expected to be up and all traffic is expected to be successful.
+  * Repeat the same test on ATE:Port2<br><br><br>
+
+   
+
+ 
+**RT-1.35.6**
+  
+```TODO: gNOI.ClearBGPNeighborRequest_GRACEFUL used in this case is under review in https://github.com/openconfig/gnoi/pull/214```
+  
+  ***`a. Expected behavior when "Administrative Reset" Notification (rfc4486) sent to the peer while the ERR policy is attached to the neighborship.`***
+   *   Start traffic as per the flows above
+   *   Trigger BGP Notification (code 6 subocde 4) from DUT:Port1 towards ATE:Port1. Please use the `gNOI.ClearBGPNeighborRequest_GRACEFUL` message.
+   *   Cease notification of Code 6, subcode 4 will result in tcp connection reset but the routes aren't flushed
+   *   Configure ATE:Port1 to not send/accept any more TCP conenctions from the DUT:Port1 until the "reset timer" on the DUT expires.
+     *   Expected behavior is the same as RT-1.35.2.a
+   *   Revert ATE configurtion to allow for the BGP sessions to be up. Restart traffic and confirm that there is zero packet loss. Expected behavior is same as the base test in RT-1.35.1
+   *   Restart the above procedure for the EBGP peering between DUT:Port-2 and ATE:Port-2<br>
+
+  
+   ***`b. Expected behavior when "Administrative Reset" Notification sent to the peer and ERR policy isnt attached.`***
+   *   Follow the same process as RT-1.35.6.a. However since the ERR policy isnt attached, expected behavior is the same as RT-1.35.2.b<br><br><br>
+
+   
+
+**RT-1.35.7**
+
+```TODO: gNOI.ClearBGPNeighborRequest_GRACEFUL used in this case is under review in https://github.com/openconfig/gnoi/pull/214```
+  
+  ***`a. Expected behavior when "Administrative Reset" Notification (rfc4486) received from the peer while ERR policy is attached on the neighborship.`***
+  * Follow the same procedure as RT-1.35.6.a above. However this time, Trigger BGP Notification (code 6 subocde 4) from ATE:Port1 towards DUT:Port1. Please use the `gNOI.ClearBGPNeighborRequest_GRACEFUL` message.
```

Comment:
Same questions as above w.r.t who is triggering the session reset. In all the cases I am reading that the DUT is doing a reset because of the reference to gNOI.

Reply:
So AIUI, if an implementation sends Administrative reset, it knows it sent Code6 subcode 4 and both the peers have negotiated "N" bit already in the RESET flag of the Capability 64 message. Therefore, while the TCP is reset, both do not flush the routes learnt from each other. Now based on the ERR policy, some routes (w/ Accept action) will experience extended retention while some routes (w/ Reject action) will be held only until the RESTART timer expires. Beyond the RESTART timer, if "STALE TIMER" timer is configured (https://datatracker.ietf.org/doc/html/rfc8538#section-4.1), the routes will stay post which they are flushed.